tei_adapter/tei_id_extractor/src/main/scala/weco/catalogue/tei/id_extractor/TeiIdExtractorWorkerService.scala

```diff
@@ -1,13 +1,105 @@
 package weco.catalogue.tei.id_extractor
 
-import uk.ac.wellcome.messaging.sns.SNSMessageSender
+import akka.stream.scaladsl.Flow
+import software.amazon.awssdk.services.sqs.model.{Message => SQSMessage}
+import uk.ac.wellcome.json.JsonUtil._
+import uk.ac.wellcome.messaging.MessageSender
+import uk.ac.wellcome.messaging.sns.NotificationMessage
 import uk.ac.wellcome.messaging.sqs.SQSStream
+import uk.ac.wellcome.storage.s3.S3ObjectLocation
+import uk.ac.wellcome.storage.store.Writable
 import uk.ac.wellcome.typesafe.Runnable
+import weco.flows.FlowOps
 
-import scala.concurrent.Future
+import java.net.URI
+import java.time.ZonedDateTime
+import scala.concurrent.{ExecutionContext, Future}
 
-case class TeiIdExtractorWorkerService(messageStream: SQSStream[Nothing],
-                                       messageSender: SNSMessageSender)
-    extends Runnable {
-  override def run(): Future[Any] = ???
+case class TeiFileChangedMessage(path: String,
+                                 uri: URI,
+                                 timeModified: ZonedDateTime)
+case class TeiIdChangeMessage(id: String,
+                              s3Location: S3ObjectLocation,
+                              timeModified: ZonedDateTime)
+case class TeiIdExtractorConfig(concurrentFiles: Int,
+                                bucket: String,
+                                teiDirectories: Set[String] = Set(
+                                  "Arabic",
+                                  "Batak",
+                                  "Egyptian",
+                                  "Greek",
+                                  "Hebrew",
+                                  "Indic",
+                                  "Javanese",
+                                  "Malay"))
```

Comment:
I can put a link to the tei repo, i think looking at that is pretty self explanatory.
The only approach that I can think of to deduce programmatically is going via a blacklist instead of a whitelist approach like here, but I think it might be a bit more fiddly? Also, I don't think new languages are added very often. 
This field could be loaded from config and everything could be moved in terraform which at least skips the pr/release/deploy bit. I don't think it's very useful doing that right now though.
I think this is something to keep in mind but IMO this is good enough for now